src/tests/tracing/eventpipe/processinfo/processinfo.cs

```diff
@@ -191,6 +195,35 @@ public static int Main(string[] args)
 
             Utils.Assert(expectedArchValue.Equals(arch), $"OS must match current Operating System. Expected: \"{expectedArchValue}\", Received: \"{arch}\"");
 
+            // VALIDATE ENV
+            // env block is an array of strings of the form "key=value" delimited by null
+            // e.g., "key=value\0key=value\0";
+            start = end;
+            end = start + 4 /* sizeof(uint32_t) */;
+            UInt32 envCount = BitConverter.ToUInt32(response.Payload[start..end]);
+            Logger.logger.Log($"envCount: {envCount}");
+
+            var env = new Dictionary<string,string>();
+            for (int i = 0; i < envCount; i++)
+            {
+                start = end;
+                end = start + 4 /* sizeof(uint32_t) */;
+                UInt32 pairLength = BitConverter.ToUInt32(response.Payload[start..end]);
+
+                start = end;
+                end = start + ((int)pairLength * sizeof(char));
+                Utils.Assert(end <= totalSize, $"String end can't exceed payload size. Expected: <{totalSize}, Received: {end} (decoded length: {pairLength})");
+                string envPair = System.Text.Encoding.Unicode.GetString(response.Payload[start..end]).TrimEnd('\0');
+                string[] parts = envPair.Split('=');
+                Utils.Assert(parts.Count() == 2, $"Malformed environment entry when splitting element on '='.  Expected: 2 parts, Received: {parts.Count()} parts {envPair}");
```

Comment:
Guess it doesn't matter, but environment variables can legally exist with empty values. You can pass empty string to ::SetEnvironmentVariableW. Then the block just contains eg `foo=\0`